Mage.Sets/src/mage/cards/c/ChissGoriaForgeTyrant.java

```diff
@@ -0,0 +1,213 @@
+        cards.retainZone(Zone.EXILED, game);
+
+        Cards castableCards = new CardsImpl(cards.getCards(StaticFilters.FILTER_CARD_NON_LAND, game));
+        Set<MageObjectReference> morSet = new HashSet<>();
+        castableCards.stream()
+                .map(uuid -> new MageObjectReference(uuid, game))
+                .forEach(morSet::add);
+
+        game.addEffect(new ChissGoriaForgeTyrantCanPlayEffect(morSet), source);
+        game.addEffect(new ChissGoriaForgeTyrantAffinityEffect(morSet), source);
+        return true;
+    }
+}
+
+class ChissGoriaForgeTyrantCanPlayEffect extends AsThoughEffectImpl {
+
+    private final Set<MageObjectReference> morSet = new HashSet<>();
+
+    ChissGoriaForgeTyrantCanPlayEffect(Set<MageObjectReference> morSet) {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.EndOfTurn, Outcome.Benefit);
+        this.morSet.addAll(morSet);
+    }
+
+    private ChissGoriaForgeTyrantCanPlayEffect(final ChissGoriaForgeTyrantCanPlayEffect effect) {
+        super(effect);
+        this.morSet.addAll(effect.morSet);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantCanPlayEffect copy() {
+        return new ChissGoriaForgeTyrantCanPlayEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        UUID objectIdToCast = CardUtil.getMainCardId(game, sourceId);
+        return game.getCard(objectIdToCast).isArtifact(game) 
+            && morSet.stream().anyMatch(mor -> mor.refersTo(objectIdToCast, game))
+            && ChissGoriaForgeTyrantWatcher.checkRef(source, morSet, game);
+    }
+}
+
+class ChissGoriaForgeTyrantAffinityEffect extends ContinuousEffectImpl {
+
+    private final Set<MageObjectReference> morSet = new HashSet<>();
+
+    public ChissGoriaForgeTyrantAffinityEffect(Set<MageObjectReference> morSet) {
+        super(Duration.Custom, Layer.AbilityAddingRemovingEffects_6, SubLayer.NA, Outcome.Benefit);
+        this.morSet.addAll(morSet);
+    }
+
+    protected ChissGoriaForgeTyrantAffinityEffect(final ChissGoriaForgeTyrantAffinityEffect effect) {
+        super(effect);
+        this.morSet.addAll(effect.morSet);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantAffinityEffect copy() {
+        return new ChissGoriaForgeTyrantAffinityEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+         if (!ChissGoriaForgeTyrantWatcher.checkRef(source, morSet, game)) {
+             discard();
+             return false;
+         }
+
+        for (Card card : game.getExile().getAllCardsByRange(game, source.getControllerId())) {
+            if (morSet.contains(new MageObjectReference(card, game))) {
```

Comment:
I didn't think of this yesterday, but could also check card type here so as not to display the ability on nonartifacts (just a cosmetic improvement)